opensrp-anc/src/main/java/org/smartregister/anc/library/fragment/MeFragment.java

```diff
@@ -48,14 +74,52 @@ protected void initializePresenter() {
 
     @Override
     protected void onViewClicked(View view) {
-        int i = view.getId();
-        if (i == R.id.logout_section) {
+        int viewId = view.getId();
+        if (viewId == R.id.logout_section) {
             DrishtiApplication.getInstance().logoutCurrentUser();
-        } else if (i == R.id.site_characteristics_section) {
+        } else if (viewId == R.id.site_characteristics_section) {
             getContext().startActivity(new Intent(getContext(), SiteCharacteristicsActivity.class));
-        } else if (i == R.id.me_pop_characteristics_section) {
+        } else if (viewId == R.id.me_pop_characteristics_section) {
             getContext().startActivity(new Intent(getContext(), PopulationCharacteristicsActivity.class));
+        } else if (viewId == R.id.language_switcher_section) {
+            languageSwitcherDialog(locales, languages);
+        }
+    }
+
+    private void languageSwitcherDialog(List<Pair<String, Locale>> locales, String[] languages) {
+        if (getActivity() != null) {
+            AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
+            builder.setTitle(getActivity().getResources().getString(R.string.choose_language));
+            builder.setItems(languages, (dialog, which) -> {
+                Pair<String, Locale> lang = locales.get(which);
+                languageSwitcherText.setText(String.format(getActivity().getResources().getString(R.string.default_language_string), lang.getLeft()));
+                LangUtils.saveLanguage(getActivity().getApplication(), lang.getValue().getLanguage());
+                Intent intent = getActivity().getIntent();
+                getActivity().finish();
+                getActivity().startActivity(intent);
+                AncLibrary.getInstance().notifyAppContextChange();
+            });
+
+            AlertDialog dialog = builder.create();
+            dialog.show();
+        }
+    }
+
+    private void registerLanguageSwitcher() {

```

Comment:
This functionality could more easily be achieved using a `Map` of languages to locales.